core/tests/protractor_utils/PreferencesPage.js

```diff
@@ -75,54 +88,47 @@ var PreferencesPage = function() {
   };
 
   this.editUserBio = async function(bio) {
-    await userBioElement.sendKeys(bio);
-    await navBar.click();
-    await preferencesLink.click();
-  };
-
-  this.get = async function() {
-    await browser.get(USER_PREFERENCES_URL);
-    await waitFor.pageToFullyLoad();
+    await action.sendKeys('User bio field', userBioElement, bio);
+    await saveNewChanges('User Bio');
   };
 
   this.toggleEditorRoleEmailsCheckbox = async function() {
-    await editorRoleEmailsCheckbox.click();
+    await action.click('Editor role emails checkbox', editorRoleEmailsCheckbox);
+    await saveNewChanges('Editor Role Emails');
   };
 
   this.toggleFeedbackEmailsCheckbox = async function() {
-    await feedbackMessageEmailsCheckbox.click();
+    await action.click(
+      'Feedback emails checkbox', feedbackMessageEmailsCheckbox);
+    await saveNewChanges('Feedback Emails');
   };
 
   this.selectSystemLanguage = async function(language) {
-    await systemLanguageSelector.click();
-    var option = element(
-      by.cssContainingText('.select2-dropdown li', language));
-    await option.click();
+    await action.select2(
+      'System language selector', systemLanguageSelector, language);
+    await saveNewChanges('System Language');
   };
 
   this.selectPreferredAudioLanguage = async function(language) {
-    await preferredAudioLanguageSelector.click();
-    var correctOption = element(
-      by.cssContainingText('.select2-results li', language));
-    await correctOption.click();
+    await action.select2(
+      'Audio language selector', preferredAudioLanguageSelector, language);
+    await saveNewChanges('Preferred Audio Language');
   };
 
   this.setUserBio = async function(bio) {
-    await waitFor.visibilityOf(
-      userBioElement, 'User bio field takes too long to appear.');
-    await userBioElement.clear();
-    await userBioElement.sendKeys(bio);
-    await navBar.click();
-    await waitFor.elementToBeClickable(
-      preferencesLink, 'Preferences takes too long to be clickable.');
-    await preferencesLink.click();
+    var inputFieldName = 'User bio input field';
+    await action.clear(inputFieldName, userBioElement);
+    await saveNewChanges('User Bio');
+    await action.sendKeys(inputFieldName, userBioElement, bio);
+    await saveNewChanges('User Bio');
   };
 
   this.setUserInterests = async function(interests) {
     await userInterestsInput.click();
-    await Promise.all(interests.map(async function(interest) {
-      await userInterestsInput.sendKeys(interest, protractor.Key.RETURN);
-    }));
+    for (var i = 0; i < interests.length; i++) {
+      await userInterestsInput.sendKeys(interests[i], protractor.Key.RETURN);
```

Comment:
Found this structure more stable i.e, earlier all the items in the list were processing in parallel, through for loop it was happening one at a time!

/cc FYI @U8NWXD @kevintab95